### DISCUSSION TITLE
Dynamic routing problem

### DISCUSSION THREAD

**Discussion by lewatt23** [vercel/next.js#12582] [Help]
Hello please i need help in nextjs , i have a blog page that  contains and index.js and [slug].js page, i want the [slug].js to capture all the "blog/[slug]" urls and the index.js to capture  the /blog url, please can someone help me out ?
ps: i don't want to create a subfolder for the slug of the articles because i want my route to  be 
www.mywebsite.com/blog/slug for the [slug].js
and
www.mywebsite.com/blog for the index.js

**Comment by timneutkens** [comment]
You can use catch-all routes:

```
pages/blog/[...slug].js
```

https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes

**Reply by lewatt23** [reply]
Thanks, please what trying to do is that by default the **/blog** open the index.js and the /blog/[slug] open the **/[slug]** page, cause as for now if i open the /blog there is a collision  between the pages.

**Reply by timneutkens** [reply]
`/blog` matches `pages/blog/index.js`
`/blog/hello-world` matches`pages/blog/[slug].js`

**Comment by lewatt23** [comment]
Hello i already tried that but as i said there are collisions, both pages are loaded on the `**/blog**`

**Reply by timneutkens** [reply]
How? That seems impossible as Next.js can't render 2 pages on 1 route. Please provide a full reproduction.
